### PR TITLE
Add sniff for deprecated Theme related WP Constants.

### PR DESCRIPTION
**Pull request by jrfnl**
Covers the sniff + unit test for issue #23.

This PR covers the currently known list as can be found in: https://github.com/Otto42/theme-check/pull/162/files

### CODE REVIEW

**Comment by jrfnl** [comment]
Updated to read 'Deprecated' not 'Forbidden'

File: WordPress/Tests/Theme/DeprecatedWPConstantsUnitTest.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Unit test class for WordPress Coding Standard.
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     https://make.wordpress.org/core/handbook/best-practices/coding-standards/
+ */
+
+/**
+ * WordPress_Tests_Theme_DeprecatedWPConstants
```
Comment (grappler) [inline WordPress/Tests/Theme/DeprecatedWPConstantsUnitTest.php]: Missing a full stop at the end here.

File: WordPress/Tests/Theme/DeprecatedWPConstantsUnitTest.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	 *
+	 * The key of the array should represent the line number and the value
+	 * should represent the number of warnings that should occur on that line.
+	 *
+	 * @return array(int => int)
+	 */
+	public function getWarningList() {
+		return array();
+
+	} // end getWarningList()
```
Comment (grappler) [inline WordPress/Tests/Theme/DeprecatedWPConstantsUnitTest.php]: Did we want to keep the closing comment?

File: WordPress/Tests/Theme/DeprecatedWPConstantsUnitTest.php (hunk 1 above)
Comment (jrfnl) [inline WordPress/Tests/Theme/DeprecatedWPConstantsUnitTest.php]: Working on improving the standard documentation in WPCS and will port back to any code in the theme review branch if/when my changes are accepted in WPCS. Ok?

**Comment by ernilambar** [comment]
@grappler `define( 'HEADER_TEXTCOLOR', '#F00' );` this code does not seem to be caught by this sniff.

**Comment by jrfnl** [comment]
@ernilambar That's correct behaviour.
This sniff is not testing for people defining Core constants as that would give a PHP error of "Constant already defined" anyway.
It is only testing for usage of the deprecated constants.

**Comment by ernilambar** [comment]
```
define('HEADER_IMAGE_WIDTH',1200);
define('HEADER_IMAGE_HEIGHT',600);
```
This type of code and similar should be caught by the sniff I believe. This code wont give `Constant already defined` error, since this was the recommended method used to override Custom Header before 3.4. And WP is good at BC even if it is lame. :D
`// Constants are lame. Don't reference them. This is just for backwards compatibility.`

**Comment by jrfnl** [comment]
In that case, another sniff should be created (or additional check added to this sniff) to check for `define()` statements and `const` statements.
That was not the original purview of this sniff _"Check for usage of deprecated WP constants"_ See https://github.com/WPTRT/WordPress-Coding-Standards/issues/23

If you think that TRTCS should also check for constants being defined, please open a new issue in this repo so this can be properly discussed.

**Pull request merged**